app/src/main/java/com/brandon/forzenbook/usecase/CreateUserUseCaseImpl.kt

```diff
@@ -15,15 +16,23 @@ class CreateUserUseCaseImpl(
         email: String,
         date: Date,
         location: String
-    ): Boolean {
+    ): CreateUserOutcome {
         val createUserData = CreateUserData(
             firstName = firstName,
             lastName = lastName,
             email = email,
             dateOfBirth = date,
             location = location
         )
-        return repository.createUser(createUserData)
+        return try {
+            val outcome = repository.createUser(createUserData)
+            if (outcome) {
```

Comment:
Instead of this `boolean` structure, just let the other errors throw and return in other catch blocks

Reply:
now have a try catch that returns based on results.